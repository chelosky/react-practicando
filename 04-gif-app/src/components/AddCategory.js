import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories  }) => {

    const [value, setValue] = useState('');

    const handleInputChange = (e) => {
        setValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if(value.trim().length > 2){
            setCategories( item => [ value, ...item ] );
            setValue('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input
                type="text"
                value={ value }
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}