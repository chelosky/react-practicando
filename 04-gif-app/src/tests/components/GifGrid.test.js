import React from "react";
import "@testing-library/jest-dom";

import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe('GifGrid', () => {

    const category = 'Dragon Ball Z';

    test('Snapshot', () => {

        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        expect(wrapper).toMatchSnapshot();
    });

    test('should  show images when load item in useFetchGifs', () => {
   
        const gifs = [{
            id: 'AAA',
            url: 'https://localhost.com/cosa.jpg',
            title: 'titulo AAA'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid category={ category } /> );
        
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('p').exists()).toBe(false);

        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
});

