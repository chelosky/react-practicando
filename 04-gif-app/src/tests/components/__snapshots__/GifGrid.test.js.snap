// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Snapshot 1`] = `
<Fragment>
  <h3>
     
    Dragon Ball Z
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid should  show images when load item in useFetchGifs 1`] = `
<Fragment>
  <h3>
     
    Dragon Ball Z
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="AAA"
      key="AAA"
      title="titulo AAA"
      url="https://localhost.com/cosa.jpg"
    />
  </div>
</Fragment>
`;
